application-ckeditor-ui/src/main/resources/CKEditor/EditSheet.xml

```diff
@@ -268,24 +268,30 @@ define('xwiki-ckeditor', [
 
   // We have to pass the plugin that makes the request (the initiator) because the expected response can be different
   // (e.g. between the filebrowser and filetools plugins).
-  var getUploadURL = function(document, initiator) {
+  var getUploadURL = function(document, initiator, allowTemporaryAttachments) {
```

Comment:
Adding the flag to the upload URL **here** means the URL needs to be updated when the configuration changes. This complicates a lot the code (I really don't like ``switchTemporaryAttachmentUpload``). I prefer to modify the ``xwiki-upload`` plugin to listen to https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_editor.html#event-fileUploadRequest and set a custom HTTP header that can be read from the ``FileUploader``.

Reply:
> means the URL needs to be updated when the configuration changes.

Do we agree that if I remove `switchTemporaryAttachmentUpload` and rely entirely on the configuration flag, as you proposed in https://github.com/xwiki-contrib/application-ckeditor/pull/50/files#r848825407 we don't need to support updating it since the call to replace would do it? I'm not sure here.

Reply:
The upload URL would be the same no matter if temporary attachments are allowed / supported or not. The ``xwiki-upload`` plugin will listen to the upload request and pas the information to the server to indicate if temporary attachments are allowed / supported. The easiest way to pass this information is through some custom HTTP headers I think. See for instance https://ckeditor.com/docs/ckeditor4/latest/guide/dev_file_upload.html#request-2 .

> You can also add custom request headers or set flags for the default request.

Reply:
So if I understand properly what you're saying, the idea is that on each `fileUploadRequest` to check the editor config and set a request header with the `allowTemporaryAttachments` flag value, then in FileUploaders use that header value (if present) to decide which upload to use. Is that correct?

Reply:
Yes